apps/s_time.c

```diff
@@ -240,16 +224,10 @@ int s_time_main(int argc, char **argv)
                                    www_path);
             if (buf_len <= 0 || SSL_write(scon, buf, buf_len) <= 0)
                 goto end;
-            while ((i = SSL_read(scon, buf, sizeof(buf))) > 0 ||
-                        SSL_get_error(scon, i) == SSL_ERROR_WANT_READ ||
-                        SSL_get_error(scon, i) == SSL_ERROR_WANT_WRITE)
```

Comment:
E.g. in the SSL_read() man page:

> If the underlying BIO is blocking, a read function will only return once the read operation has been finished or an error occurred, except when a renegotiation takes place, in which case a SSL_ERROR_WANT_READ may occur.

That text probably needs some updating for TLSv1.3. Historically it could also occur if a heartbeat message is received (but that's been removed now). I think(?) it will also occur if a warning alert is received. Basically if any record arrives that doesn't yield application data.